### PR TITLE
Updated Annotations with new Rego subheading and example

### PR DESCRIPTION
**Pull request by Parsifal-M**
Added Rego subheading to the annotations documentation section and added an example of using annotations, then below that a link to the Rego policy reference.

This is for #4937

Signed-off-by: Peter Macdonald <macdonald.peter90@gmail.com>

### CODE REVIEW

File: docs/content/annotations.md [hunk 1]
```diff
@@ -261,6 +261,48 @@ allow {
...
+		"message": annotation.description,
+	}
+}
+```
+
+will output
+
+```live:example/metadata/1:output
+```
+If you'd like more examples and information on this, you can see more here under the [Rego](../policy-reference/#rego) policy reference.
```
Comment (srenatus) [inline docs/content/annotations.md:305]: [nit] Let's add a newline for separation here. It won't matter for markup, though.

**Review by srenatus — APPROVED** [review]

**Review by srenatus — COMMENTED** [review]
Let's also link to the other section as mentioned in the issue, perhaps?

File: docs/content/annotations.md (hunk 1 above)
Comment (Parsifal-M) [reply docs/content/annotations.md:305]: Sure! No worries

**Comment by Parsifal-M** [comment]
> Let's also link to the other section as mentioned in the issue, perhaps?

Is that not `If you'd like more examples and information on this, you can see more here under the [Rego](../policy-reference/#rego) policy reference.` Unless I misunderstood the link @anderseknert mentions is https://www.openpolicyagent.org/docs/latest/policy-reference/#rego

Please let me know? Thanks!

**Review by srenatus — APPROVED** [review]
Of course. Sorry my bad. LGTM

**Comment by Parsifal-M** [comment]
Hey @srenatus 

No worries, did not mean to "dismiss" your review, I did add the new line, think I need to resolve the conversation before I push again sorry for that :sweat_smile: 

Thanks!

**Comment by srenatus** [comment]
No worries, github does that because it's configured to do it. This isn't on you 😉 Thanks for contributing! 🥳

**Pull request merged**